### PR TITLE
fix(datasource):  convert the type of ob-mysql-sharding data source to ob-mysql

### LINKED ISSUE #1252: [Bug]: The back-end DialectType has added ODP sharding, which may cause the front-end page to crash
### ODC version

4.2.3

### OB version

odp sharding mysql

### What happened?

The back-end `DialectType` has added ODP sharding MySQL, which may cause the front-end page to crash.

### What did you expect to happen?
There is no ODP sharding mysql in the DialectType returned by the backend to the frontend.

### How can we reproduce it (as minimally and precisely as possible)?
This hidden bug has not been triggered so far

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by PeachThinking**
#### What type of PR is this?
type-bug
module-datasource

#### What this PR does / why we need it:
The back-end `DialectType` has added `ODP_SHARDING_OB_MYSQL`, which may cause the front-end page to crash. This hidden bug has not been triggered so far.
In order to avoid this bug, `ODP_SHARDING_OB_MYSQL` is converted to `OB_MYSQL` during serialization.

#### Which issue(s) this PR fixes:

Fixes #1252

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java [hunk 1]
```diff
@@ -171,4 +182,23 @@ public OBDatabaseUser getSysTenantUser(String instanceId) {
...
+    /**
+     * In order to adapt to the fact that there is no ODP_SHARDING_OB_MYSQL in DialectType of the
+     * front-end, ODP_SHARDING_OB_MYSQL is converted to OB_MYSQL during serialization.
+     */
+    private static class DialectTypeOutputSerializer extends JsonSerializer<DialectType> {
+
+        @Override
+        public void serialize(DialectType dialectType, JsonGenerator jsonGenerator,
+                SerializerProvider serializerProvider) throws IOException {
+            if (Objects.nonNull(dialectType)) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:194]: what if null?

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java [hunk 2]
```diff
@@ -67,9 +74,13 @@ public PasswordEncoder passwordEncoder() {
     public ObjectMapper objectMapper(SensitivePropertyHandler sensitivePropertyHandler) {
         CustomOutputSerializer customOutputSerializer = new CustomOutputSerializer()
                 .addSerializer(Internationalizable.class, new I18nOutputSerializer());
+        DialectTypeOutputSerializer dialectTypeOutputSerializer = new DialectTypeOutputSerializer();
+        SimpleModule dialectTypeModule =
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:78]: `SimpleModule dialectTypeModule = new SimpleModule().addSerializer(DialectType.class, new DialectTypeOutputSerializer());` may be better

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java [hunk 3]
```diff
@@ -171,4 +182,23 @@ public OBDatabaseUser getSysTenantUser(String instanceId) {
         }
     }
 
+    /**
+     * In order to adapt to the fact that there is no ODP_SHARDING_OB_MYSQL in DialectType of the
+     * front-end, ODP_SHARDING_OB_MYSQL is converted to OB_MYSQL during serialization.
+     */
+    private static class DialectTypeOutputSerializer extends JsonSerializer<DialectType> {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:189]: add `@Deprecated` annotation to notice others that this is a temporary solution

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java (hunk 2 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:77]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java (hunk 3 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:190]: done

File: server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java (hunk 1 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/BeanConfiguration.java:195]: Added logic for handling null ：`jsonGenerator.writeNull();`

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**